rust/src/builder.rs

```diff
@@ -61,37 +61,28 @@ macro_rules! impl_buffer_builder {
                 (byte_capacity / mem::size_of::<$native_ty>()) as i64
             }
 
-            /// Push a value into the builder, growing the internal buffer as needed
+            /// Pushes a value into the builder, growing the internal buffer as needed.
             pub fn push(&mut self, v: $native_ty) -> Result<()> {
                 self.reserve(1)?;
                 self.write_bytes(v.to_byte_slice(), 1)
             }
 
-            /// Push a slice of type T, growing the internal buffer as needed
+            /// Pushes a slice of type `T`, growing the internal buffer as needed.
             pub fn push_slice(&mut self, slice: &[$native_ty]) -> Result<()> {
                 let array_slots = slice.len() as i64;
                 self.reserve(array_slots)?;
                 self.write_bytes(slice.to_byte_slice(), array_slots)
             }
 
-            /// Reserve memory for n elements of type T
+            /// Reserves memory for `n` elements of type `T`.
             pub fn reserve(&mut self, n: i64) -> Result<()> {
                 let new_capacity = self.len + n;
-                if new_capacity > self.capacity() {
-                    return self.grow(new_capacity);
-                }
-                Ok(())
-            }
-
-            /// Grow the internal buffer to `new_capacity`, where `new_capacity` is the capacity in
-            /// elements of type T
-            fn grow(&mut self, new_capacity: i64) -> Result<()> {
                 let byte_capacity = mem::size_of::<$native_ty>() * new_capacity as usize;
-                self.buffer.resize(byte_capacity)?;
+                self.buffer.reserve(byte_capacity)?;
                 Ok(())
             }
 
-            /// Build an immutable `Buffer` from the existing internal `MutableBuffer`'s memory
+            /// Consumes this and returns an immutable `Buffer`.
```

Comment:
Is there a missing word here?  i.e. Consumes this `BufferBuilder` and...

Reply:
Done.